src/persist-client/src/impl/machine.rs

```diff
@@ -39,6 +40,7 @@ use crate::ShardId;
 pub struct Machine<K, V, T, D> {
     consensus: Arc<dyn Consensus + Send + Sync>,
     metrics: Arc<Metrics>,
+    collector: GarbageCollector,
```

Comment:
nit: other parts call this `gc`, but ... :man_shrugging: I don't care too much

Reply:
oh did I miss some? I originally named it "gc" but swapped to make it match "compacter" because I couldn't figure out a way to write "gcer". maybe I'll just rename compacter to compact!